src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PassSystemUriObjectsInsteadOfStrings.cs

```diff
@@ -24,15 +29,191 @@ public abstract class PassSystemUriObjectsInsteadOfStringsAnalyzer : DiagnosticA
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Usage,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182360.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            // this is stateless analyzer, can run concurrently
+            analysisContext.EnableConcurrentExecution();
+
+            // this has no meaning on running on generated code which user can't control
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            analysisContext.RegisterCompilationStartAction(c =>
+            {
+                var analyzer = new Analyzer(c.Compilation, GetInvocationExpression);
+                c.RegisterOperationAction(analyzer.Analyze, OperationKind.InvocationExpression);
+            });
+        }
+
+        protected abstract SyntaxNode GetInvocationExpression(SyntaxNode invocationNode);
+
+        private struct Analyzer
+        {
+            private static readonly string[] uriWords = new string[] { "uri", "urn", "url" };
+
+#pragma warning disable RS1008 // this type will be created per compilation
```

Comment:
This seems like a bug in RS1008, it probably only ignores nested types. You should file an issue to clean up RS1008 here.